### PR TITLE
Classifier: remove default drawing tool gestures

### LINKED ISSUE #2960: Classifier: Drawing Task's base "Mark" may need default .initialDrag() et al
## Classifier Functionality Bug

Package: `lib-classifier`
Affects: Drawing Tasks

The Drawing Task has a base "Drawing Mark" code (drawingTools/models/marks/Mark/Mark.js) that is extended into different drawing marks, e.g. Circle Mark, Polygon Mark, etc.

There are a few bits of code that assume EVERY type of Mark has certain _default methods,_ but problem is, this isn't true in every case. Some "default methods" are actually _only available in the extended Marks_ (like Circle Marks) but _not the default Mark._

For example, in the base "Drawing Tool" code (drawingTools/models/tools/Tool/Tool.js), there's this code...

```
handlePointerMove(event, mark) {
  mark.initialDrag(event)
},
```

...which causes problems when an extended Mark doesn't explicitly include the missing "default" `initialDrag()`. (e.g., the Polygon Mark)

Context: this bug was discovered by me and @ErikOstlund alongside #2959 . This problem probably would have gone undetected if we didn't stumble into 2959's extremely niche trigger condition which allowed us to have mismatched Marks and Tools, so good news is that this problem probably isn't being experienced by any actual users at the moment.

### Dev Notes

Suggested solution is simple: add an empty `initialDrag(event) { /* Do nothing */ }` to Mark.js

I'm **not sure** if the following things _also_ need defaults in the base Mark.js code, but I suspect the answer is _probably yes_ for two of them:
- `mark.initialPosition()` - used in InteractionLayer
- `mark.move()` - used in InteractionLayer 
- `mark.setCoordinates()` - ehhh? Not sure if this should apply to all marks.

### Status

Low priority. The niche trigger conditions means this only affects FEM projects that use the Polygon Tool, but it's good practice anyway to have default fallbacks since the code already assumes those defaults exist.

### PR DESCRIPTION
**Pull request by eatyourgreens**
Drawing tools were set up with a default behaviour that's common to all tools: create a new mark on drag ( `mark.initialPosition(e)` on `pointerDown` and `mark.initialDrag(e)` on `pointerMove`) then finish it on release (`mark.finish()` on `pointerUp`.) This single gesture works for points, lines, circles, rectangles and other simple shapes but doesn't work for extended shapes that require more than one gesture. Complex shapes (transcription lines, ellipses, polygons and freehand lines, at the moment) have to override the default gesture before we can add new code.

This removes the default event handlers from the `Tool` component and copies them explicitly into each drawing tool. Tool code is more verbose but there's no hidden, default tool behaviour to trip up developers of new tools.

With the changes here, ellipse tools can be configured so that they are created by two drag gestures, not one. This means that any subtask popups wait until the ellipse is complete before opening.

https://user-images.githubusercontent.com/59547/178510388-6ba2a878-228c-4fd1-99d2-c210e620a0a5.mov

Closes #2960.

## Package
lib-classifier

## How to Review
I have a test workflow that uses a variety of drawing tools.
https://localhost:8080/?project=908&workflow=3370&language=en

I Fancy Cats, `/projects/brooke/i-fancy-cats` on staging, has an ellipse with a popup question, which can be used to test the changes to the Ellipse tool.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js [hunk 1]
```diff
@@ -113,8 +113,7 @@ function InteractionLayer({
     }
 
     if (creating) {
-      activeTool.handlePointerDown &&
-        activeTool.handlePointerDown(convertEvent(event), activeMark)
+        activeTool?.handlePointerDown?.(convertEvent(event), activeMark)
```
Comment (shaunanoordin) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js:116]: [very minor][code style] The indentation here is a bit off, by 2 spaces.

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js [hunk 2]
```diff
@@ -136,8 +135,7 @@ function InteractionLayer({
 
   function onPointerUp(event) {
     if (creating) {
-      activeTool.handlePointerUp &&
-        activeTool.handlePointerUp(convertEvent(event), activeMark)
+        activeTool?.handlePointerUp?.(convertEvent(event), activeMark)
```
Comment (shaunanoordin) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js:138]: [very minor][code style] The indentation here is also a bit off, by 2 spaces.

**Review by shaunanoordin — APPROVED** [review]
## PR Review

Package: `lib-classifier`
Affects: Drawing Tasks

This PR is a refactor of multiple drawing tools with minimal UI/behaviour changes.

- Code read is good: 
  - The responsibility for _implementing_ `someTool.handlePointerUp()` and `someTool.handlePointerMove()` has been moved from the general Tool class to the specific sub-tools.
  - Accordingly, InteractionLayer now has existential safeties `?.` when calling handlePointerUp/Move. 
- New Behaviour: for the Ellipse tool, if it has a Subtask, the Subtask popup only appears after the user completes both Step 1: drag to set x-radius, and Step 2: drag to set y-radius.
  - Note: Subtask Popup continues to appear normally when clicking to select an existing Ellipse.

### Testing

Tested on localhost with macOS + Chrome

Test Setup:
- A [test WF](https://local.zooniverse.org:8080/?env=staging&project=darkeshard%2Ftest-project-2022&workflow=3596) with a Drawing Task, with multiple Drawing Tools

Testing Steps:
- Place, edit, and delete multiple types of Drawing Tools
  - Points: click to create ; click-drag to create & reposition ; drag to reposition
  - Rectangle: click-drag to create ; drag to reposition ; drag corners to resize
  - Circle: click-drag to create (drag for radius) ; drag to reposition ; drag radius handle to resize
  - Polygon: click to place nodes ; click on first node to finish ; drag nodes on existing polygon to reshape (reposition node); drag edges on existing polygon to reposition
  - Ellipse: click-drag to create (drag for x-radius), then 🆕 drag y-radius handle to set y-radius _and complete_ ; drag to reposition ; drag x/y-radius handle set x/y-radius and/or rotate.

Testing LGTM 👍 

### Status

LGTM, let's go 👍

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js (hunk 1 above)
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js:116]:
```suggestion
      activeTool?.handlePointerDown?.(convertEvent(event), activeMark)
```

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js (hunk 2 above)
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js:138]:
```suggestion
      activeTool?.handlePointerUp?.(convertEvent(event), activeMark)
```

**Pull request merged**